specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-02-01-preview/onPremiseSensors.json

```diff
@@ -0,0 +1,352 @@
+      "post": {
+        "x-ms-examples": {
+          "Download file for reset password of the sensor": {
+            "$ref": "./examples/OnPremiseSensors/DownloadResetPassword.json"
+          }
+        },
+        "tags": [
+          "Sensors"
+        ],
+        "operationId": "OnPremiseSensors_DownloadResetPassword",
+        "description": "Download file for reset password of the sensor",
+        "produces": [
+          "application/zip"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/OnPremiseSensorName"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "required": true,
+            "description": "Input for reset password.",
+            "schema": {
+              "$ref": "#/definitions/ResetPasswordInput"
+            },
+            "x-ms-parameter-location": "method"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Downloaded reset password file",
+            "schema": {
+              "type": "object",
+              "format": "file"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "OnPremiseSensorsList": {
+      "description": "List of on-premise IoT sensors",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List data",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OnPremiseSensor"
+          }
+        }
+      }
+    },
+    "OnPremiseSensor": {
+      "type": "object",
+      "description": "On-premise IoT sensor",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "On-premise IoT sensor properties",
+          "$ref": "#/definitions/OnPremiseSensorProperties"
+        },
+        "systemData": {
+          "readOnly": true,
+          "type": "object",
+          "description": "Azure Resource Manager metadata containing createdBy and modifiedBy information.",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "OnPremiseSensorProperties": {
```

Comment:
This resource is represents a non-Azure resource (on-premise deployment of our sensor).
Currently we have no need for additional properties in this resource. In case that would change in the future, we'll add it.